src/index.tsx

```diff
@@ -1065,10 +1061,17 @@ export class App extends React.Component<any, AppState> {
                     resetSelection();
                   },
                 });
-                this.setState({
-                  elementType: "selection",
-                  editingElement: element,
-                });
+                if (!this.state.elementLocked) {
+                  resetCursor();
+                  this.setState({
+                    editingElement: element,
+                    elementType: "selection",
+                  });
+                } else {
+                  this.setState({
+                    editingElement: element,
+                  });
+                }
```

Comment:
This one actually causes a UX regression: when you have locked shapes, and create a text element, clicking away to finalize the text creates a new one.

Reply:
I've fixed this in https://github.com/excalidraw/excalidraw/pull/746/commits/f157e419a7f3876d61374caf7d09f39cb330663b